src/factory/service.ts

```diff
@@ -32,6 +40,7 @@ export interface ICDCService {
     client: DBClient,
     database: string,
     resource: string,
+    mqueueClient: KafkaProducerCompact<AzureResource>,
```

Comment:
`AzureResource` could not be the correct type. Since this is an interface it should be a generic document I guess.